File: frontend/src/api/followups.ts
import client from './client'

export interface FollowUp {
  id: string
  chat_id: string
  follow_up_question: string
  message_summary: string
  scheduled_for: string
  urgency: 'normal' | 'urgent'
  status: 'pending' | 'sent' | 'dismissed' | 'resolved'
  created_at: string
  chats?: { title: string }
}

export async function fetchFollowUps(): Promise<{ follow_ups: FollowUp[] }> {
  const { data } = await client.get('/follow-ups')
  return data
}

export async function dismissFollowUp(id: string): Promise<void> {
  await client.post(`/follow-ups/${id}/dismiss`)
}

export async function resolveFollowUp(id: string): Promise<void> {
  await client.post(`/follow-ups/${id}/resolve`)
}

File: backend/api/routes/followups.py
from fastapi import APIRouter, Depends, HTTPException

from backend.api.dependencies import get_current_user
from backend.services.followup_service import (
    get_pending_follow_ups,
    dismiss_follow_up,
    resolve_follow_up,
)

router = APIRouter()


@router.get("/follow-ups")
async def list_follow_ups(user_id: str = Depends(get_current_user)):
    items = await get_pending_follow_ups(user_id)
    return {"follow_ups": items}


@router.post("/follow-ups/{follow_up_id}/dismiss")
async def dismiss(follow_up_id: str, user_id: str = Depends(get_current_user)):
    await dismiss_follow_up(follow_up_id, user_id)
    return {"status": "dismissed"}


@router.post("/follow-ups/{follow_up_id}/resolve")
async def resolve(follow_up_id: str, user_id: str = Depends(get_current_user)):
    await resolve_follow_up(follow_up_id, user_id)
    return {"status": "resolved"}
